src/elements/e-video.vue

```diff
@@ -0,0 +1,201 @@
+      :src="mapSrc"
+      :allowfullscreen="allowfullscreen"
+      :frameborder="frameborder.toString()"
+      :width="width.toString()"
+      :height="height.toString()"
+    ></iframe>
+    <!-- </div> -->
+  </div>
+</template>
+
+<script lang="ts">
+  import { PropType, defineComponent } from 'vue';
+
+  const videoUrlSources = {
+    youtube: 'https://www.youtube.com',
+    vimeo: 'https://vimeo.com/',
+  };
+
+  const playerUrlSources = {
+    youtube: 'https://www.youtube.com/embed/',
+    vimeo: 'https://player.vimeo.com/video/',
+  };
+
+  export default defineComponent({
+    name: 'e-video',
+    status: 0, // TODO: remove when component was prepared for current project.
+
+    // components: {},
+
+    props: {
+      /**
+       * Allows the user to watch the video in fullscreen
+       */
+      allowfullscreen: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Specifies the source of the video
+       *
+       * We can decide between youtube and vimeo
+       */
+      source: {
+        type: String as PropType<'youtube' | 'vimeo'>,
+        required: true,
+        validator(value: string) {
+          return Object.keys(playerUrlSources).includes(value);
+        },
+      },
+
+      /**
+       * Allows to add/remoe the border around the player
+       *
+       * When set to 0 the border is completely removed
+       *
+       * Default from iframe is 2
+       */
+      frameborder: {
+        type: [String, Number],
+        default: 2,
+      },
+
+      /**
+       * Accepts a valid url from a youtube or vimeo video
+       */
+      videoUrl: {
+        type: String,
+        default: null,
+      // validate against different sources
+      // validator: (value: string) => playerSources[this.source].includes(value)
+      // validator: (value: string) => {
+      //   return Object.values(videoUrlSources).includes(value);
+      // }
+      },
+
+      /**
+       * Accepts an ID from a youtube or vimeo video
+       */
+      videoId: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Allows to set the video player width.
+       */
+      width: {
+        type: [String, Number],
+        default: 560,
+      },
+
+      /**
+       * Allows to set the video player height.
+       */
+      height: {
+        type: [String, Number],
+        default: 315,
+      },
+
+    /**
+     * Defines the players size to be responsive 100% width to the parent container
+     */
+    // responsive: {
+    //   type: Boolean,
+    //   default: null
+    // }
+    },
+
+    // emits: {},
+    // setup(): Setup {},
+
+    computed: {
+      mapSrc(): string | undefined {
+        const playerBaseUrl = playerUrlSources[this.source];
+
+        if (!(this.videoId || this.videoUrl)) {
+          // eslint-disable-next-line no-console
+          console.error("Neither 'videoId' or 'videoUrl' were defined", this.$el);
+
+          return undefined;
```

Comment:
Do we need to return undefined or could we just use return and make the return type string optional? As I am new to typescript myself I don't know if this is a good convention or not :D

Reply:
If we disable ESlint `consistent-return` rule, we can just use `return`. 
As far as I know, having an optional return type with the ? operator is not possible in TS.

`return string | undefined` is the "official" way to type an optional return type

Reply:
hint: this is fine with me. In most cases we use `null` instead of `undefined` but that probably needs to change with the use of TS.

Reply:
According to the TS coding guidelines undefined is the way to go.
https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined